docs/docs/creating-a-local-plugin.md

```diff
@@ -4,24 +4,68 @@ title: Creating a Local Plugin
 
 If a plugin is only relevant to your specific use-case, or if you’re developing a plugin and want a simpler workflow, a locally defined plugin is a convenient way to create and manage your plugin code.
 
+## Project structure for a local plugin
+
 Place the code in the `plugins` folder in the root of your project like this:
 
-```text
-plugins
-└── my-own-plugin
-    └── package.json
+```
+/my-gatsby-site
+└── gatsby-config.js
+└── /src
+└── /plugins
+    └── /my-own-plugin
+        └── package.json
+```
+
+The plugin also needs to be added to your `gatsby-config.js`, because there is no auto-detection of plugins. It can be added alongside any other 3rd party Gatsby plugins already included in your config.
+
+For the plugin to be discovered when you run `gatsby develop`, the plugin's root folder name needs to match the name used in the `gatsby-config.js` (_not_ its _name_ in its `package.json` file). For example, in the above structure, the correct way to load the plugin is:
+
+```javascript:title=gatsby-config.js
+module.exports = {
+  plugins: [
+    `gatsby-third-party-plugin`,
+    `my-own-plugin`, // highlight-line
+  ],
+}
 ```
 
-**NOTE:** You still need to add the plugin to your `gatsby-config.js`. There is no auto-detection of local plugins.
+Then your plugin can begin to hook into Gatsby through Node and SSR APIs.
```

Comment:
```suggestion
Then the plugin can begin to hook into Gatsby through Node and SSR APIs.
```
Not sold on this one. Seems like it could go either way?